core/txpool/legacypool/legacypool.go

```diff
@@ -1430,24 +1427,6 @@ func (pool *LegacyPool) reset(oldHead, newHead *types.Header) {
 	pool.addTxsLocked(reinject)
 }
 
-// reduceBalanceByL1Cost returns the given balance, reduced by the L1Cost of the first transaction in list if applicable
-// Other txs will get filtered out necessary.
-func (pool *LegacyPool) reduceBalanceByL1Cost(list *list, balance *uint256.Int) *uint256.Int {
-	if !list.Empty() && pool.l1CostFn != nil {
-		el := list.txs.FirstElement()
-		if l1Cost := pool.l1CostFn(el.RollupCostData()); l1Cost != nil {
-			l1Cost256 := uint256.MustFromBig(l1Cost)
-			if l1Cost256.Cmp(balance) >= 0 {
-				// Avoid underflow
-				balance = uint256.NewInt(0)
```

Comment:
What's the overall strategy with underflows / overflows? Seems like in the new code we usually `panic`, but I notice in the old code here we would just clamp to `0`.

Reply:
Underflows must never happen if the accounting is correct, so it doesn't make sense to clamp. This would be a critical error. We always subtract something that was added before.

Overflows are checked, e.g. in `Add`, and return an error.